### PR TITLE
Display friends activities

### PR DESCRIPTION
**Pull request by JoniDutoit**
## Description

This PR aims to display the user's friends activities on the community tab of the application. Merging this PR will allow the user to check the most recent friends activities easily, being able to check what they have been up to lately! 

## Note
Although it's not the aim of this PR to touch the user's profile screen, it was essential since fetching the user's activities was firstly designed for the profile screen and testing getting user's activities from the database  throught the profile screen makes life much simpler! 

## Modifications
- Feed from Community Tab now displays user's friends most recent activities (setted to 7 days)
- Profile Screen is now renabled to fetch current user's activties
- Added a new class ActivityConverter.kt to manage conversion from data models and database

## Video

### CODE REVIEW

File: app/src/main/java/com/lastaoutdoor/lasta/ui/screen/social/components/FriendsActivities.kt [hunk 1]
```diff
@@ -48,39 +59,96 @@ fun FriendsActivityList(isConnected: ConnectionState, latestFriendActivities: Li
 }
 
 @Composable
-fun FriendsActivityCard(activity: UserActivity) {
+fun FriendsActivityCard(friendActivity: FriendsActivities) {
```
Comment (Andy130604) [inline app/src/main/java/com/lastaoutdoor/lasta/ui/screen/social/components/FriendsActivities.kt:62]: Nice way of making modularised test tags!

File: app/src/main/java/com/lastaoutdoor/lasta/viewmodel/SocialViewModel.kt [hunk 2]
```diff
@@ -51,7 +52,7 @@ constructor(
   var hasFriendRequest: Boolean = friendRequests.isNotEmpty()
 
   // number of days to consider for the latest activities
-  private val numberOfDays = 7
+  private val timeFrame = TimeFrame.W
```
Comment (Andy130604) [inline app/src/main/java/com/lastaoutdoor/lasta/viewmodel/SocialViewModel.kt:55]: This is really consistent work removing some hard coded constants

**Review by Andy130604 — APPROVED** [review]
This is a masterclass! Love your new feed. Everything looks really nice. I just think you could use a bit more documentation on your backend code which could come in helpful for the next sprint.

**Comment by Thimphou** [comment]
Great new code! I'll definitely use your way of modularising test tags in the future :)

**Pull request merged**